drivers/char/tpm/tpm_ftpm_optee.c

```diff
@@ -76,37 +81,35 @@ static int ftpm_optee_tpm_op_send(struct tpm_chip *chip, u8 *buf, size_t len)
 		{ .attr = TEE_IOCTL_PARAM_ATTR_TYPE_NONE }
 	};
 	struct tee_ioctl_invoke_arg submit_command_args;
-	struct tee_shm *shm = NULL;
 
-	mutex_lock(&tpm_tee_context.response_buffer_lock);
+	if (tpm_tee_context.response_length != 0) {
+		rc = -EIO;
```

Comment:
It seems like `-EBUSY` may be better suited, if I understand the code correctly: you can't send a command if the response buffer has data in it. If so, then this isn't an unexpected I/O error, rather the developer using the TPM messed up.

Reply:
I am not sure if I agree. I am no expert on these error codes, so if you feel strongly, I can change it. 

I can send a command, but I do not think it is right. So I am explicitly failing here. And it is not because I am waiting on the TPM to respond (which would be EBUSY), but it is because send/recv (I/O) operations are out of order. That's why EIO. Thoughts?

Reply:
Mm... I suppose the rationale behind suggesting `-EBUSY` is that it's not that the TPM itself is doing anything, but rather there is a pending response buffer which hasn't yet been dealt with. You know best what's going on under the hood, so if you think `-EIO` is correct, that's fine by me.